server/memphis_handlers_stations.go

```diff
@@ -685,17 +686,18 @@ func (sh StationsHandler) CreateStation(c *gin.Context) {
 	}
 
 	var retentionType string
-	if body.RetentionType != "" && body.RetentionValue > 0 {
+	if body.RetentionType != "" {
 		retentionType = strings.ToLower(body.RetentionType)
 		err = validateRetentionType(retentionType)
 		if err != nil {
 			serv.Warnf("CreateStation: Station " + body.Name + ": " + err.Error())
 			c.AbortWithStatusJSON(SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": err.Error()})
 			return
 		}
-	} else {
-		retentionType = "message_age_sec"
-		body.RetentionValue = 604800 // 1 week
+	}
+
+	if retentionType == "infinite" {
```

Comment:
I am not sure about this hack. I've added it to ensure that i will not break any logic that rely to this field 😅